File: LeetCode/Python Solutions/Adding Spaces to a String/add.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        # Convert spaces to set for O(1) lookup
        spaces_set = set(spaces)
        result = []
        
        for i in range(len(s)):
            if i in spaces_set:
                result.append(' ')
            result.append(s[i])
            
        return ''.join(result)

File: LeetCode/Java Solutions/Adding Spaces to a String/add.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        StringBuilder result = new StringBuilder();
        int spaceIndex = 0;
        
        for (int i = 0; i < s.length(); i++) {
            // Add space if current index is in spaces array
            if (spaceIndex < spaces.length && i == spaces[spaceIndex]) {
                result.append(' ');
                spaceIndex++;
            }
            result.append(s.charAt(i));
        }
        
        return result.toString();
    }
}